gapis/replay/wait_for_fence.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2019 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package replay
+
+import (
+	"context"
+
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/api/transform"
+	"github.com/google/gapid/gapis/replay/builder"
+)
+
+type WaitForFence struct{}
+
+func (t *WaitForFence) Transform(ctx context.Context, id api.CmdID, cmd api.Cmd, out transform.Writer) {
+	if cmd.CmdFlags(ctx, id, out.State()).IsStartOfFrame() || id == 0 {
```

Comment:
Is this going to insert a wait event at the start of every frame?

Is this what we want?

Reply:
After talking with @chrisforbes It seems like we should only insert it at the beginning of the replay.  I'll remove the `IsStartOfFrame` check unless we think we want to profile each frame individually.

Reply:
SGTM. We can always adjust this later..